sormas-ui/src/main/java/de/symeda/sormas/ui/person/PersonController.java

```diff
@@ -192,6 +195,62 @@ private void mergePersons(PersonGrid personGrid, List<PersonIndexDto> personInde
 		List<EventParticipantSelectionDto> eventParticipantsWithSameEvent =
 			FacadeProvider.getEventParticipantFacade().getEventParticipantsWithSameEvent(firstPersonUuid, secondPersonUuid);
 
+		Map<String, Map<String, Integer>> eventParticipantsByEvent = new HashMap<>();
+		Set<String> eventsWithDuplicatePersonEventParticipants = new HashSet<>();
+		eventParticipantsWithSameEvent.forEach(eventparticipant -> {
+			if (eventParticipantsByEvent.containsKey(eventparticipant.getEvent().getUuid())) {
+				Map<String, Integer> personsByEvent = eventParticipantsByEvent.get(eventparticipant.getEvent().getUuid());
+				if (personsByEvent.containsKey(eventparticipant.getPersonUuid())) {
+					personsByEvent.put(eventparticipant.getPersonUuid(), personsByEvent.get(eventparticipant.getPersonUuid()) + 1);
+					eventParticipantsByEvent.put(eventparticipant.getEvent().getUuid(), personsByEvent);
+					eventsWithDuplicatePersonEventParticipants.add(eventparticipant.getEvent().getUuid());
+				} else {
+					eventParticipantsByEvent.get(eventparticipant.getEvent().getUuid()).put(eventparticipant.getPersonUuid(), 1);
+				}
+			} else {
+				Map<String, Integer> eventParticipantByPersonCount = new HashMap<>();
+				eventParticipantByPersonCount.put(eventparticipant.getPersonUuid(), 1);
+				eventParticipantsByEvent.put(eventparticipant.getEvent().getUuid(), eventParticipantByPersonCount);
+			}
+		});
+
+		if (!eventsWithDuplicatePersonEventParticipants.isEmpty()) {
+
+			VerticalLayout component = new VerticalLayout();
+			component.addComponent(
+				VaadinUiUtil
+					.createInfoComponent(I18nProperties.getString(Strings.infoPickorMergeEventParticipantDuplicateEventParticipantByPersonByEvent)));
+
+			eventsWithDuplicatePersonEventParticipants.forEach(eventUuid -> {
+				Label eventUuidLabel = new Label(eventUuid);
+				component.addComponent(eventUuidLabel);
+			});
+
+			CommitDiscardWrapperComponent<VerticalLayout> eventsWithDuplicateEventParticipants = new CommitDiscardWrapperComponent<>(component);
+
+			eventsWithDuplicateEventParticipants.getButtonsPanel()
```

Comment:
should work like: `eventsWithDuplicateEventParticipants.getCommitButton().setVisible(false)`

but does it make sense to hide it if at the end of the method you replaces this button with a cancel button?

Reply:
the component was changed